tests/unit/auth_handler.cpp

```diff
@@ -0,0 +1,724 @@
+// Copyright 2022 Memgraph Ltd.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.txt; by using this file, you agree to be bound by the terms of the Business Source
+// License, and you may not use this file except in compliance with the Business Source License.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0, included in the file
+// licenses/APL.txt.
+
+#include <gflags/gflags.h>
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include "auth/auth.hpp"
+#include "auth/models.hpp"
+#include "glue/auth_handler.hpp"
+#include "query/typed_value.hpp"
+#include "utils/file.hpp"
+#include "utils/rw_lock.hpp"
+#include "utils/synchronized.hpp"
+
+class AuthQueryHandlerFixture : public testing::Test {
+ protected:
+  std::filesystem::path test_folder_{std::filesystem::temp_directory_path() / "MG_tests_unit_auth_handler"};
+  memgraph::utils::Synchronized<memgraph::auth::Auth, memgraph::utils::WritePrioritizedRWLock> auth{
+      test_folder_ / ("unit_auth_handler_test_" + std::to_string(static_cast<int>(getpid())))};
+  memgraph::glue::AuthQueryHandler auth_handler{&auth, memgraph::glue::default_user_role_regex.data()};
+
+  std::string user_name = "Mate";
+  std::string edge_type_repr = "EdgeType1";
+  std::string label_repr = "Label1";
+  memgraph::auth::Permissions perms{};
+  memgraph::auth::FineGrainedAccessHandler handler{};
+
+  virtual void SetUp() {
+    memgraph::utils::EnsureDir(test_folder_);
+    memgraph::utils::license::global_license_checker.EnableTesting();
+  }
+
+  virtual void TearDown() {
+    std::filesystem::remove_all(test_folder_);
+    perms = memgraph::auth::Permissions{};
+    handler = memgraph::auth::FineGrainedAccessHandler{};
+  }
+};
+
+TEST_F(AuthQueryHandlerFixture, GivenAuthQueryHandlerWhenInitializedHaveNoUsernamesOrRolenames) {
+  ASSERT_FALSE(auth_handler.GetUsernames().size());
+  ASSERT_FALSE(auth_handler.GetRolenames().size());
+}
+
+TEST_F(AuthQueryHandlerFixture, GivenUserWhenNoDeniesOrGrantsThenNothingIsReturned) {
+  memgraph::auth::User user = memgraph::auth::User{user_name, "", perms, handler};
+  auth->SaveUser(user);
+
+  { ASSERT_TRUE(auth_handler.GetUsernames().size()); }
+
+  {
+    auto privileges = auth_handler.GetPrivileges(user_name);
+
+    ASSERT_FALSE(privileges.size());
```

Comment:
Consider using `ASSERT_EQ`, because then it will print out the actual value in case of a failure. By implicitly converting `size_t` to `bool`, it will only print `true` for `false`.